### ISSUE TITLE
Investigate later steps

### ISSUE THREAD

**Issue by Hukyl** [dl4cv-ai-pinnacle/rpi-face-recognition#2]
# Investigate Next Steps

Here is a general list of steps to investigate next. Feel free to add to it or suggest something else.

1. **Phase 1: Program Minimum (Analytical Pipeline)**

    - [ ] Run ready-made edge models on Raspberry Pi
    - [ ] Benchmark FPS, latency, memory for different model combinations
    - [ ] Report performance comparison across edge models

2. **Phase 2: Model Conversion & Quantization Experiments**

    Based on the conversion flow: **PyTorch/TF → .safetensors/.bin/.model → ONNX → .tf → TFLite**

    <img width="500" alt="Image" src="https://github.com/user-attachments/assets/f5859f0b-765b-4010-a92e-456eaba2533d" />

    - [ ] Test TFLite runtime on Raspberry Pi (primary target format)
    - [ ] Explore ExecuTorch as alternative PyTorch edge runtime
    - [ ] Convert models between formats, measure accuracy loss at each step:
        - PyTorch (.safetensors, .bin) → ONNX → TF → TFLite
        - TF (.model, .tf weights-only) → TFLite directly
    - [ ] Compare quantization levels (FP32 → FP16 → INT8 → FP8):
        - Accuracy degradation per level
        - Speed/size gains per level

3. **Phase 3: Optional Fine-tuning (If Time Permits)**

    - [ ] Fine-tune detection models specifically for face + hand signs
    - [ ] Accept accuracy trade-off on other object classes
    - [ ] Quantization-aware training for better INT8 performance

**Notes from Kuzmenko**:

- Pure analytical comparison is valid project scope
- Fine-tuning not strictly necessary, but possible for domain-specific tasks
- TFLite is recommended path for Raspberry Pi deployment
